### PR TITLE
Fix RKE2 binary path check

### PR DESCRIPTION
**Pull request by jLemmings**
# Description

When upgrading a RKE2 cluster which either /usr/local is its own partition or its not writeable then it fails to use the rke2 binary. A similar fix has been made before for the hardening part. This PR moves the partition check to an earlier step and uses the information deciding where the binary is located.

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Small minor change not affecting the Ansible Role code (GitHub Actions Workflow, Documentation etc.)

## How Has This Been Tested?
Locally

### CODE REVIEW

**Comment by MonolithProjects** [comment]
@jLemmings thanks for the PR. Could you please resolve the conflicts?

**Comment by jLemmings** [comment]
@MonolithProjects fixed the new merge conflicts.

File: tasks/rke2.yml [hunk 1]
```diff
@@ -31,59 +30,59 @@
...
+        timeout: 30
+    - name: Download RKE2 artifacts and compare with checksums
+      ansible.builtin.get_url:
+        url: "{{ rke2_artifact_url }}/{{ rke2_version }}/{{ item }}"
+        dest: "{{ rke2_artifact_path }}/{{ item }}"
+        mode: 0640
+        checksum: "sha256:{{ rke2_artifact_url }}/{{ rke2_version }}/sha256sum-{{ rke2_architecture }}.txt"
+        timeout: 30
+      with_items: "{{ rke2_artifact | reject('search', 'sha256sum') | list }}"
+    rescue:
```
Comment (MonolithProjects) [inline tasks/rke2.yml:48]: Please fix the indentation

**Review by MonolithProjects — CHANGES_REQUESTED** [review]

File: tasks/rke2.yml (hunk 1 above)
Comment (jLemmings) [reply tasks/rke2.yml]: Already had it fixed in my other branch and chose the other branch to merge here... Should be good to go now.

**Review by MonolithProjects — APPROVED** [review]
LGTM

**Pull request merged**